src/net_processing.cpp

```diff
@@ -780,6 +780,14 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
         PushNodeVersion(pnode, connman, GetTime());
 }
 
+void PeerLogicValidation::QueueUnbroadcastTxs(CNode* pnode)
+{
+    for(const uint256& txid : mempool.m_unbroadcast_txids){
+        CInv inv(MSG_TX, txid);
+        pnode->PushInventory(inv);
```

Comment:
How do you know that it won't be rejected due to "filterInventoryKnown"?
What is the rationale here?

Reply:
It might. Which upholds the same guarantees as the previous logic it's replacing.

On current master, wallet transactions get rebroadcast through a scheduler flow that makes its way to `ResendWalletTransactions`, which selects transactions to rebroadcast then invokes `SubmitMemoryPoolAndRelay`. This goes to `BroadcastTransaction`, which checks ATMP then relays via `RelayTransaction`, which uses the `CNode#PushInventory` function that only adds the hash if its not contained in `filterInventoryKnown`. 

This PR changes the initial parts of the flow... so `BroadcastTransaction` adds the transaction to `mempool.m_unbroadcast_txids`, then there is a scheduled job to periodically `AttemptInitialBroadcast`, but the behavior is matched where the locally-submitted transactions would be reattempted to send to the network, filtered for the connections that were already tried, and sent out to new connections.

Changing that behavior could be desirable for robustness but of course reduces privacy. So out of scope for this PR. If you think it may be desirable, I'd be happy to think it through / talk about it more, we can open a new issue. (I'm working on the assumption that current delivery guarantees are sufficient).

Reply:
Ah I see. Just want to be clear that if the network is not really used, the bloom filter doesn't roll and transaction may be not rebroadcast in days.
If that's what you have in mind when designing the whole thing, let's just leave a comment that this can happen.

Reply:
The `filterInventoryKnown` is per-peer though, so it just won't be rebroadcast to a peer that already knew about it somehow; but it'll stay in the unbroadcast set and be retried for other peers until a getdata actually makes its way through (or until it gets dropped from the mempool)? If all of your peers ignore your INV and don't request a GETDATA, you'll not send more INVs until you find a new peer, but... even if you did, those peers would probably still not send a GETDATA anyway, so what else would you do?

Reply:
Maybe it's the naming that confuses me. It's called "rebroadcast", not "consider to rebroadcast".
>even if you did, those peers would probably still not send a GETDATA anyway
Not sure about that. If transaction expires from a mempool, and there's not much activity on-chain, filter would still prevent us from re-announcing it. The correspondence between the two is questionable.

But yeah, if that's only my own confusion, I don't care about just keeping it this way without any comments.

Reply:
> Maybe it's the naming that confuses me. It's called "rebroadcast", not "consider to rebroadcast".

the language I'm introducing is "unbroadcast", which means consider it as initial broadcast. but there are these interactions with `filterInventoryKnown` to consider in different cases. 

I added some comments [here](https://github.com/bitcoin/bitcoin/pull/18038/files#diff-eff7adeaec73a769788bb78858815c91R782) in an attempt to make this more apparent. 

going to resolve this conversation, but @naumenkogs let me know if you think this is something we need to more carefully consider in regards to this PR.